tools/accuracy_checker/accuracy_checker/evaluators/custom_evaluators/custom_text_recognition_evaluator.py

```diff
@@ -264,13 +274,78 @@ def __init__(self, network_info, launcher, models_args, meta, is_blob=None):
         })
         if not contains_all(network_info, ['recognizer_encoder', 'recognizer_decoder']):
             raise ConfigError('network_info should contain encoder and decoder fields')
-        self.vocab = meta['vocab']
         self.recognizer_encoder = create_recognizer(network_info['recognizer_encoder'], launcher, 'encoder')
         self.recognizer_decoder = create_recognizer(network_info['recognizer_decoder'], launcher, 'decoder')
         self.sos_index = 0
         self.eos_index = 2
         self.max_seq_len = int(network_info['max_seq_len'])
 
+    def get_phrase(self, indices):
+        raise NotImplementedError()
+
+    def predict(self, identifiers, input_data):
+        raise NotImplementedError()
+
+    def reset(self):
+        pass
+
+    def release(self):
+        self.recognizer_encoder.release()
+        self.recognizer_decoder.release()
+
+
+class SequentialTextRecognitionModel(BaseSequentialModel):
+    def __init__(self, network_info, launcher, models_args, meta, is_blob=None):
+        super().__init__(network_info, launcher, models_args, meta, is_blob=is_blob)
+        self.vocab = network_info['custom_label_map']
+
+    def get_phrase(self, indices):
+        res = ''
+        for idx in indices:
+            if idx != self.eos_index:
+                res += str(self.vocab.get(idx, '?'))
+            else:
+                break
+        return res
+
+    def predict(self, identifiers, input_data):
+        assert len(identifiers) == 1
+        input_data = np.array(input_data)
+        input_data = np.transpose(input_data, (0, 3, 1, 2))
+        enc_res = self.recognizer_encoder.predict(
+            inputs={'imgs': input_data})
+        features = enc_res['features']
+        dec_state = enc_res['decoder_hidden']
+
+        tgt = np.array([[self.sos_index]])
+        logits = []
+        for _ in range(self.max_seq_len):
+
+            dec_res = self.recognizer_decoder.predict(inputs={'features': features,
+                                                              'hidden': dec_state,
+                                                              'decoder_input': tgt,
+                                                              })
+
+            dec_state = dec_res['decoder_hidden']
+            logit = dec_res['decoder_output']
+            logits.append(logit)
+            tgt = np.array([[np.argmax(np.array(logit), axis=1)]])
```

Comment:
why do we need double square brackets here?

Reply:
removed